llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -2710,6 +2710,23 @@ Value *InstCombiner::getFreelyInvertedImpl(Value *V, bool WillInvertAllUses,
   return nullptr;
 }
 
+static bool isJointMatrixTargetExtTy(Type *Ty) {
+  return Ty->isTargetExtTy() &&
+         Ty->getTargetExtName() == "spirv.JointMatrixINTEL";
```

Comment:
I agree with @jsji. This doesn't look like a good solution.
We should avoid customization of LLVM passes in intel/llvm. Changes to passes like InstCombine should be pushed to llvm.org.
Having said that, I believe that LLVM passes are supposed to handle "canonical" LLVM, so adjusting OpenCL CPU compiler sounds reasonable.

Reply:
+1 to @jsji and @bader